sklearn/utils/_weight_vector.pyx.tp

```diff
@@ -106,9 +110,11 @@ cdef class WeightVector{{name_suffix}}(object):
             val = x_data_ptr[j]
             innerprod += (w_data_ptr[idx] * val)
             xsqnorm += (val * val)
+            l1norm_delta += val
             w_data_ptr[idx] += val * (c / wscale)
 
         self.sq_norm += (xsqnorm * c * c) + (2.0 * innerprod * wscale * c)
+        self.l1_norm += l1norm_delta * (c / wscale)
```

Comment:
This looks like it doesn't take the weights (w_data_ptr) into account because the innerprod term is missing?